File: 2025/04 April/23 Unique Number II/main.py
# Link: https://www.geeksforgeeks.org/problems/finding-the-numbers0215/1


class Solution:
    def singleNum(self, arr):
        freq = {}
        for num in arr:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
        return sorted([num for num in freq if freq[num] == 1])


if __name__ == "__main__":
    sol = Solution()
    arr = [1, 2, 3, 2, 1, 4]
    result = sol.singleNum(arr)
    print(" ".join(map(str, result)))

    arr = [2, 1, 3, 2]
    result = sol.singleNum(arr)
    print(" ".join(map(str, result)))

    arr = [2, 1, 3, 3]
    result = sol.singleNum(arr)
    print(" ".join(map(str, result)))

File: 2025/04 April/23 Unique Number II/main.cpp
// Link: https://www.geeksforgeeks.org/problems/finding-the-numbers0215/1

#include <iostream>
#include <vector>
#include <map>
using namespace std;

class Solution
{
public:
    vector<int> singleNum(vector<int> &arr)
    {
        map<int, int> mpp;
        for (auto it : arr)
            mpp[it]++;
        vector<int> v;
        for (auto it : mpp)
        {
            if (it.second == 1)
                v.push_back(it.first);
        }
        return v;
    }
};

int main()
{
    Solution sol;
    vector<int> arr = {1, 2, 3, 2, 1, 4};
    vector<int> result = sol.singleNum(arr);
    for (int i : result)
        cout << i << " ";
    cout << endl;

    arr = {2, 1, 3, 2};
    result = sol.singleNum(arr);
    for (int i : result)
        cout << i << " ";
    cout << endl;
    arr = {2, 1, 3, 3};
    result = sol.singleNum(arr);
    for (int i : result)
        cout << i << " ";
    cout << endl;
    return 0;
}